### PR TITLE
Add email validation

### PR DESCRIPTION
**Pull request by konstyefremov**
https://github.com/happy-travel/agent-app-project/issues/453

### CODE REVIEW

File: Api/Services/Invitations/AgentInvitationCreateService.cs [hunk 1]
```diff
@@ -105,12 +105,27 @@ void LogInvitationCreated()
...
+                .Ensure(IsInvitationEmailValid, "Invitation email is not valid")
+                .Bind(CreateInvitation)
                 .Check(SendInvitationMailPipe);
 
 
             Task<Result> SendInvitationMailPipe(string invitationCode) 
                 => SendInvitationMail(invitationCode, prefilledData, invitationType, inviterAgencyId);
+
+
+            async Task<bool> IsInvitationEmailValid()
```
Comment (mingazhev) [inline Api/Services/Invitations/AgentInvitationCreateService.cs:118]: Please use something like GenericValidator to validate e-mail address with the existing instruments

Comment (konstyefremov) [reply Api/Services/Invitations/AgentInvitationCreateService.cs]: done

File: Api/Services/Invitations/AgentInvitationCreateService.cs [hunk 2]
```diff
@@ -105,12 +106,30 @@ void LogInvitationCreated()
...
+                    v.RuleFor(x => x.UserRegistrationInfo.FirstName).NotEmpty();
+                    v.RuleFor(x => x.UserRegistrationInfo.LastName).NotEmpty();
+                    v.RuleFor(x => x.UserRegistrationInfo.Title).NotEmpty();
+                    v.RuleFor(e => e.UserRegistrationInfo.Email).NotEmpty().EmailAddress();
+                }, prefilledData);
+            }
+
+            Task<Result<string>> CreateInvitation()
+            {
+                return Create(prefilledData, invitationType, inviterUserId, inviterAgencyId);
```
Comment (DFSko) [inline Api/Services/Invitations/AgentInvitationCreateService.cs:131]: You can use arrow function here

File: Api/Services/Invitations/AgentInvitationCreateService.cs [hunk 3]
```diff
@@ -110,26 +109,24 @@ void LogInvitationCreated()
...
+                        .WithMessage("Valid email is required");
                 }, prefilledData);
-            }
 
-            Task<Result<string>> CreateInvitation()
-            {
-                return Create(prefilledData, invitationType, inviterUserId, inviterAgencyId);
-            }
+
+            Task<Result<string>> CreateInvitation() => Create(prefilledData, invitationType, inviterUserId, inviterAgencyId);
```
Comment (DFSko) [inline Api/Services/Invitations/AgentInvitationCreateService.cs:129]: arrow function always on new line

File: Api/Services/Invitations/AgentInvitationCreateService.cs [hunk 4]
```diff
@@ -97,20 +98,35 @@ async Task<Result<UserInvitation>> SaveInvitation()
             }
 
 
-            void LogInvitationCreated()
-                => _logger.LogInvitationCreated(invitationType, prefilledData.UserRegistrationInfo.Email);
+            void LogInvitationCreated() => _logger.LogInvitationCreated(invitationType, prefilledData.UserRegistrationInfo.Email);
```
Comment (mingazhev) [inline Api/Services/Invitations/AgentInvitationCreateService.cs:101]: Arrow to the next line, by our code style

**Review by DFSko — APPROVED** [review]

**Pull request merged**